client/components/Applications/ApplicationForm.jsx

```diff
@@ -95,27 +100,63 @@ export default createForm('application', class extends Component {
     );
   }
 
-  renderCustomURLField = (application) => {
+  renderCustomURLCheckbox = () => {
+    const { fields } = this.props;
+
+    return this.isNotCustomApp() ? (
+      <InputCheckBox
+        field={fields.customURLEnabled}
+        fieldName="customURLEnabled"
+        label="Custom URL"
+        ref="customURLEnabled"
+      />
+    ) : null;
+  }
+
+  renderCustomURLField = () => {
     if (!this.props.fields.customURLEnabled.value) {
       return null;
     }
 
+    const label = this.isNotCustomApp() ? '' : 'URL';
+
     return (
       <InputText
-        field={this.props.fields.customURL} fieldName="customURL" label="" ref="customURL"
+        field={this.props.fields.customURL} fieldName="customURL" label={label} ref="customURL"
         placeholder="Add your customer URL here which will be invoked when users click the icon."
       />
     );
   }
 
+  componentDidUpdate() {
+    const { fields, clients } = this.props;
+
+    // Enable custom URL if it's a custom app
+    if ((!this.isNotCustomApp()) && !fields.customURLEnabled.value) {
+      fields.customURLEnabled.onChange(true);
+    }
+
+    // Change app name when client changes
+    if ((typeof fields.name.value === 'undefined') && fields.client.value) {
```

Comment:
This might possibly overwrite a value that I already have in there. We should copy the value over when you focus the textbox, not when you choose a different client.